### ISSUE TITLE
Stuck trying to upgrade to Transformer 2

### ISSUE THREAD

**Issue by CodySwannGT** [aws-amplify/amplify-cli#10268]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.13.1

### Amplify CLI Version

8.0.1

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

no

### Amplify Categories

api

### Amplify Commands

push

### Describe the bug

Hi, we were trying to upgrade from GraphQL Transformer 1 to 2.

During the initial deploy, we got the error that our security token had expired.

Upon trying to redeploy, we now get this error:

````
An error occurred during the push operation: ["Index: 0 State: {\"deploy\":\"deploying\"} Message: Parameters: [OpenSearchEBSVolumeGB, OpenSearchStreamBatchSize, OpenSearchAccessIAMRoleName, OpenSearchStreamingLambdaHandlerName, OpenSearchStreamMaximumBatchingWindowInSeconds, OpenSearchInstanceCount, OpenSearchInstanceType, OpenSearchDebugStreamingLambda, OpenSearchStreamingIAMRoleName, OpenSearchStreamingFunctionName, OpenSearchStreamingLambdaRuntime] do not exist in the template"]
````

### Expected behavior

We don't get that error

### Reproduction steps

I don't really know how to include this, but:

1. Upgrade to graphqltransformer2
2. amplify push
3. security expiration token error
4. amplify push 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by CodySwannGT** [comment]
This is a complete production blocker, so any feedback is hugely appreciated.

**Comment by CodySwannGT** [comment]
@josefaidt some more information here:

We've now been able to replicate this on another sandbox environment. 

1. We cloned the production environment and deployed as transformer v1 via ``amplify push``
2. We migrated the schema.graphql file to v2
3. We deployed via ``amplify push``
4. Deploy makes it to the very end. Fails and begins trying to rollback. 
5. We get: ``security token had expired`` error
6. Rollback stops (More on this below)
7. Try to redeploy via ``amplify push`` - get an error message that a deploy is in progress ([described here](https://github.com/aws-amplify/amplify-cli/issues/9665))
8. Delete deploy-state.json from s3 bucket
9. Try to redeploy via ``amplify push`` 
10. Get this error: ``An error occurred during the push operation: ["Index: 0 State: {\"deploy\":\"deploying\"} Message: Parameters: [OpenSearchEBSVolumeGB, OpenSearchStreamBatchSize, OpenSearchAccessIAMRoleName, OpenSearchStreamingLambdaHandlerName, OpenSearchStreamMaximumBatchingWindowInSeconds, OpenSearchInstanceCount, OpenSearchInstanceType, OpenSearchDebugStreamingLambda, OpenSearchStreamingIAMRoleName, OpenSearchStreamingFunctionName, OpenSearchStreamingLambdaRuntime] do not exist in the template"]``

At this point, we are completely dead in the water. The remote environment does not work and there is nothing we can do to fix it.

Please understand, we are trying to migrate to v2 because we are [having this issue](https://github.com/aws-amplify/amplify-cli/issues/6729).

Not only has this killed two of our sandbox environments which is going to cost us untold hours to reimport the data into (since Amplify doesn't have a mechanism for importing/exporting data), but we have zero confidence to try this process on our production environment given that one hiccup during migration can complete kill the environment.

As for step 6 above, in our particular case, what caused the deploy to fail was the step in the deployment process where Amplify tries to delete the auto created AmplifyDataStore table. We have a permission boundary setup that prevents amplify from deleting tables.

And, yes, we can certainly disable that permission boundary (although we setup that permission boundary because we had a massive issue where Amplify deleted production Tables, so I really don't want to).

However, going back to the above, who knows if something else will happen when running this on production and then we're stuck with a broken app

**Comment by lazpavel** [comment]
Hi @CodySwannGT, can you share the contents of `cli.json`, at the first glance looks that the transformer version rolled back to V1 while the parameters remained the ones for V2, in V1 all the above parameters have `Elastic` instead of `Open`

**Comment by CodySwannGT** [comment]
@lazpavel yes. that's exactly right. Somehow (I have no idea how). I got the one environment "cody" to complete deployment to v2.

But then, I checked out a v1 environment, did work, deployed that (as v1). Then switched back to the v2 environment (did some work) and tried to deploy and got pretty much the exact opposite error.

````
UPDATE_FAILED               apipropswap                  AWS::CloudFormation::Stack Tue Apr 26 2022 10:31:39 GMT-0400 (Eastern Daylight Time) Parameters: [ElasticSearchInstanceCount, ElasticSearchEBSVolumeGB, CreateAPIKey, ElasticSearchInstanceType] do not exist in the template
````

Attached is our entire amplify directory. To say we're desperate for a fix would be an understatement

[amplify.zip](https://github.com/aws-amplify/amplify-cli/files/8564133/amplify.zip)

**Comment by lazpavel** [comment]
Hi @CodySwannGT, please sent the amplify directory contents to amplify-cli@amazon.com, we can work to schedule a call to investigate the issue

**Comment by CodySwannGT** [comment]
@lazpavel done. Thank you!

**Comment by giulioambrogi** [comment]
Any update or workaround here? I'm completely stuck
